src/io/trailer.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License
+
+#pragma once
+#include "io/memory_io.h"
+#include "knowhere/version.h"
+
+#define KNOWHERE_TRAILER_SIZE 512  // bytes
+#define MAX_INDEX_NAME_SIZE 100    // bytes
+#define TRAILER_VALID_FLAG UINT32_C(0x0F0F0F0F)
+#define TRAILER_OFFSET(size) size - KNOWHERE_TRAILER_SIZE
+
+namespace knowhere {
+union Trailer {
+    uint8_t bytes[KNOWHERE_TRAILER_SIZE];
+    struct Meta {
+        uint32_t flag;
+        uint64_t bin_size;
+        IndexVersion version;
+        uint32_t checksum;
+        char index_name[MAX_INDEX_NAME_SIZE + 1];
+    } meta;
+    Trailer() {
+        meta.flag = TRAILER_VALID_FLAG;
+    };
+    bool
+    TrailerValidCheck() {
+        return meta.flag == TRAILER_VALID_FLAG;
+    }
+    std::string
+    GetIndexName() {
+        return std::string(meta.index_name);
+    }
+    int32_t
+    GetVersion() {
+        return meta.version;
+    }
+    uint64_t
+    GetIndexBinarySize() {
+        return meta.bin_size;
+    }
+    uint32_t
+    GetCheckSum() {
+        return meta.checksum;
+    }
+    void
+    SetIndexName(std::string index_name) {
+        if (index_name.size() > MAX_INDEX_NAME_SIZE) {
+            LOG_KNOWHERE_ERROR_ << "the size of index name larger than " << MAX_INDEX_NAME_SIZE;
```

Comment:
we should not just silently ignore the error. we should throw exception or return false so the caller of this method is aware of the error and can handle it.

Reply:
updated